code/game/atoms.dm

```diff
@@ -432,53 +660,77 @@
 		user.active_storage.show_to(user)
 	return TRUE
 
+///Get the best place to dump the items contained in the source storage item?
 /atom/proc/get_dumping_location(obj/item/storage/source,mob/user)
 	return null
 
-//This proc is called on the location of an atom when the atom is Destroy()'d
+/**
+  * This proc is called when an atom in our contents has it's Destroy() called
+  * 
+  * Default behaviour is to simply send COMSIG_ATOM_CONTENTS_DEL
+  */
 /atom/proc/handle_atom_del(atom/A)
 	SEND_SIGNAL(src, COMSIG_ATOM_CONTENTS_DEL, A)
 
-//called when the turf the atom resides on is ChangeTurfed
+/**
+  * called when the turf the atom resides on is ChangeTurfed
+  *
+  * Default behaviour is to loop through atom contents and call their HandleTurfChange() proc
+  */
 /atom/proc/HandleTurfChange(turf/T)
 	for(var/a in src)
 		var/atom/A = a
 		A.HandleTurfChange(T)
 
-//the vision impairment to give to the mob whose perspective is set to that atom (e.g. an unfocused camera giving you an impaired vision when looking through it)
+/**
+  * the vision impairment to give to the mob whose perspective is set to that atom
+  *
+  * (e.g. an unfocused camera giving you an impaired vision when looking through it)
+  */
 /atom/proc/get_remote_view_fullscreens(mob/user)
 	return
 
-//the sight changes to give to the mob whose perspective is set to that atom (e.g. A mob with nightvision loses its nightvision while looking through a normal camera)
+/**
+  * the sight changes to give to the mob whose perspective is set to that atom
+  * 
+  * (e.g. A mob with nightvision loses its nightvision while looking through a normal camera)
+  */
 /atom/proc/update_remote_sight(mob/living/user)
 	return
 
 
-//Hook for running code when a dir change occurs
+/**
+  * Hook for running code when a dir change occurs
+  *
+  * Not recommended to use, listen for the COMSIG_ATOM_DIR_CHANGE signal instead (sent by this proc)
+  */
 /atom/proc/setDir(newdir)
 	SEND_SIGNAL(src, COMSIG_ATOM_DIR_CHANGE, dir, newdir)
 	dir = newdir
 
+///Handle melee attack by a mech
 /atom/proc/mech_melee_attack(obj/mecha/M)
 	return
 
-//If a mob logouts/logins in side of an object you can use this proc
+/**
+  * Called when the atom log's in or out
+  * 
+  * Default behaviour is to call on_log on the location this atom is in
+  */
 /atom/proc/on_log(login)

```

Comment:
every single atom in the game calls on_log on it's entire contents recursively when a client logs into an atom, so that 
```

/obj/structure/bodycontainer/on_log(login)
    ..()
    update_icon()
```

that's it, nothing else, just so morgues can update their goddamn icon.

anyway i'm gonna fix that by sending a signal instead that morgues can listen for when an item is placed in side them.

Reply:
Why not just check contents.len of morgue when someone close it?